libraries/AP_AHRS/AP_AHRS_DCM.cpp

```diff
@@ -674,6 +674,11 @@ AP_AHRS_DCM::drift_correction(float deltat)
         _position_offset_north = 0;
         _position_offset_east = 0;
 
+        if (!home_is_set() && !_home_alt_set_from_gps) {
+            _home.alt = _gps.location().alt;
```

Comment:
Wouldn't it be easier to save this in a `_last_alt` variable and then use it when home isn't set?